src/org/wso2/tg/jenkins/executors/TestExecutor.groovy

```diff
@@ -90,31 +92,41 @@ def getTestExecutionMap(parallel_executor_count) {
-                        }
-                        runtime.unstashTestPlansIfNotAvailable("${props.WORKSPACE}/testplans")
-                        if (executor == parallelExecCount) {
-                            for (int i = processFileCount * (executor - 1); i < files.length; i++) {
-                                /*IMPORTANT: Instead of using 'i' directly in your logic below,
-                                you should assign it to a new variable and use it.
-                                (To avoid same 'i-object' being refered)*/
-                                // Execution logic
-                                int fileNo = i
-                                testplanId = commonUtils.getTestPlanId("${props.WORKSPACE}/test-plans/"
-                                        + files[fileNo].name)
-                                runPlan(files[i], testplanId)
+                        sh """
+                            \$(aws ecr get-login --no-include-email)
+                        """
+                    }
+                    docker.image(config.getPropertyFromTestgridConfig("TG_DOCKER_URI"))
+                            .inside("-v ${props.TESTGRID_HOME}:${props.TESTGRID_HOME} " +
+                            "-e AWS_ACCESS_KEY_ID -e AWS_DEFAULT_REGION -e AWS_SECRET_ACCESS_KEY") {
+                        script {
+                            int processFileCount = 0
+                            if (files.length < parallelExecCount) {
+                                processFileCount = 1
+                            } else {
+                                processFileCount = files.length / parallelExecCount
                             }
-                        } else {
-                            for (int i = 0; i < processFileCount; i++) {
-                                int fileNo = processFileCount * (executor - 1) + i
-                                testplanId = commonUtils.getTestPlanId("${props.WORKSPACE}/test-plans/"
-                                        + files[fileNo].name)
-                                runPlan(files[fileNo], testplanId)
+                            runtime.unstashTestPlansIfNotAvailable("${props.WORKSPACE}/testplans")
+                            if (executor == parallelExecCount) {
+                                for (int i = processFileCount * (executor - 1); i < files.length; i++) {
+                                    /*IMPORTANT: Instead of using 'i' directly in your logic below,
+                            you should assign it to a new variable and use it.
+                            (To avoid same 'i-object' being refered)*/
+                                    // Execution logic
+                                    int fileNo = i
+                                    testplanId = commonUtils.getTestPlanId("${props.WORKSPACE}/test-plans/"
+                                            + files[fileNo].name)
+                                    runPlan(files[i], testplanId)
```

Comment:
Here we are using i it seems. 

Also, can you elaborate what is meant by 'To avoid same 'i-object' being refered'?

Reply:
As per my previous tests, the `i` value was being passed by reference here.
https://github.com/groovy/Scriptom/wiki/Passing-Values-by-Reference-(in-out)